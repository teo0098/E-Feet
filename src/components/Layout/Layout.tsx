import { Navigation } from '@components/molecules';

type LayoutProps = {
  children: React.ReactNode;
  styles?: { [key: string]: string };
  className?: string;
};

const Layout: React.VFC<LayoutProps> = ({ children, className = '', styles = {} }) => {
  const classNames: string = ['min-h-screen bg-slate-100', className].join(' ');

  return (
    <div style={styles} className={classNames} data-testid="mainLayout">
      {children}
      <Navigation />
    </div>
  );
};

export default Layout;
