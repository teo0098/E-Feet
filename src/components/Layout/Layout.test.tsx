import { render, screen, within } from '@testing-library/react';

import { Layout } from '@components/Layout';

describe('Layout component tests', () => {
  it('Renders content', () => {
    render(
      <Layout>
        <h1>E-feet</h1>
      </Layout>
    );
    const navElement = screen.getByRole('navigation');
    expect(navElement).toBeVisible();
    const linkElements = within(navElement).getAllByRole('link');
    expect(linkElements).toHaveLength(5);
  });

  it('Renders with custom styles', () => {
    render(
      <Layout styles={{ color: 'red' }}>
        <h1>E-feet</h1>
      </Layout>
    );
    const layoutElement = screen.getByTestId('mainLayout');
    expect(layoutElement).toBeVisible();
    expect(layoutElement).toHaveStyle({ color: 'red' });
  });

  it('Renders with custom className', () => {
    render(
      <Layout className="text-neutral-200">
        <h1>E-feet</h1>
      </Layout>
    );
    const layoutElement = screen.getByTestId('mainLayout');
    expect(layoutElement).toBeVisible();
    expect(layoutElement).toHaveClass('text-neutral-200');
  });

  it('Snapshots content', () => {
    const { container } = render(
      <Layout>
        <h1>E-feet</h1>
      </Layout>
    );
    expect(container).toMatchSnapshot();
  });
});
