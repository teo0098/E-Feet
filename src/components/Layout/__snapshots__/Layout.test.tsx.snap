// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout component tests Snapshots content 1`] = `
<div>
  <div
    class="min-h-screen bg-slate-100 "
    data-testid="mainLayout"
  >
    <h1>
      E-feet
    </h1>
    <nav
      class="flex fixed bottom-0 left-0 justify-between p-2 w-full bg-slate-300 shadow-lg "
    >
      <a
        class="text-neutral-900 border-b-4 border-transparent "
        href="/"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="HomeRoundedIcon"
          focusable="false"
          style="font-size: 1.875rem; color: rgb(23, 23, 23);"
          viewBox="0 0 24 24"
        >
          <path
            d="M10 19v-5h4v5c0 .55.45 1 1 1h3c.55 0 1-.45 1-1v-7h1.7c.46 0 .68-.57.33-.87L12.67 3.6c-.38-.34-.96-.34-1.34 0l-8.36 7.53c-.34.3-.13.87.33.87H5v7c0 .55.45 1 1 1h3c.55 0 1-.45 1-1z"
          />
        </svg>
      </a>
      <a
        class="text-neutral-900 border-b-4 border-transparent "
        href="/search"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="SearchRoundedIcon"
          focusable="false"
          style="font-size: 1.875rem; color: rgb(23, 23, 23);"
          viewBox="0 0 24 24"
        >
          <path
            d="M15.5 14h-.79l-.28-.27c1.2-1.4 1.82-3.31 1.48-5.34-.47-2.78-2.79-5-5.59-5.34-4.23-.52-7.79 3.04-7.27 7.27.34 2.8 2.56 5.12 5.34 5.59 2.03.34 3.94-.28 5.34-1.48l.27.28v.79l4.25 4.25c.41.41 1.08.41 1.49 0 .41-.41.41-1.08 0-1.49L15.5 14zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
          />
        </svg>
      </a>
      <a
        class="text-neutral-900 border-b-4 border-transparent "
        href="/wishlist"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="FavoriteRoundedIcon"
          focusable="false"
          style="font-size: 1.875rem; color: rgb(23, 23, 23);"
          viewBox="0 0 24 24"
        >
          <path
            d="M13.35 20.13c-.76.69-1.93.69-2.69-.01l-.11-.1C5.3 15.27 1.87 12.16 2 8.28c.06-1.7.93-3.33 2.34-4.29 2.64-1.8 5.9-.96 7.66 1.1 1.76-2.06 5.02-2.91 7.66-1.1 1.41.96 2.28 2.59 2.34 4.29.14 3.88-3.3 6.99-8.55 11.76l-.1.09z"
          />
        </svg>
      </a>
      <a
        class="text-neutral-900 border-b-4 border-transparent "
        href="/cart"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="LocalMallRoundedIcon"
          focusable="false"
          style="font-size: 1.875rem; color: rgb(23, 23, 23);"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 6h-2c0-2.76-2.24-5-5-5S7 3.24 7 6H5c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V8c0-1.1-.9-2-2-2zm-7-3c1.66 0 3 1.34 3 3H9c0-1.66 1.34-3 3-3zm0 10c-2.33 0-4.29-1.59-4.84-3.75-.17-.63.32-1.25.97-1.25.47 0 .85.34.98.8.35 1.27 1.51 2.2 2.89 2.2s2.54-.93 2.89-2.2c.13-.46.51-.8.98-.8.65 0 1.13.62.97 1.25C16.29 11.41 14.33 13 12 13z"
          />
        </svg>
      </a>
      <a
        class="text-neutral-900 border-b-4 border-transparent "
        href="/user"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="PersonRoundedIcon"
          focusable="false"
          style="font-size: 1.875rem; color: rgb(23, 23, 23);"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v1c0 .55.45 1 1 1h14c.55 0 1-.45 1-1v-1c0-2.66-5.33-4-8-4z"
          />
        </svg>
      </a>
    </nav>
  </div>
</div>
`;
